aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h

```diff
@@ -93,11 +93,14 @@ struct conv_param_t {
       assert("Failed to initialize QNNPACK conv_param_t struct.");
     }
 
-    if (kernel_scale <= 0.0f || !std::isnormal(kernel_scale)) {
+    //if (kernel_scale <= 0.0f || !std::isnormal(kernel_scale)) {
+    // This has to be checked for all the output channel requant scales.
+    // TODO
+    if (requantization_scales[0] <= 0.0f || !std::isnormal(requantization_scales[0])) {
```

Comment:
I think we can check this when we create the requant scale

Reply:
I think this is better here because the constraints come from qnnpack backend.

Reply:
It would be nice if the calculation of requantization scale also happened on the backend side like before. Did you move it into the op so that we calculate it just once?

Reply:
I think earlier it was calculated during requantization param creation. It we keep it there we have to populate that array every time we run the op. It is little less efficient. Plus it does seem natural that it is captured as part of the conv_params.